samcli/commands/deploy/__init__.py

```diff
@@ -35,19 +37,197 @@
+            resource_type = resource_props.get("ResourceType")
+            changes[action].append({"LogicalResourceId": logical_id, "ResourceType": resource_type})
+    return changes
 
 
-def do_cli(args, template_file, stack_name):
-    args = args + ('--stack-name', stack_name)
+def do_cli(args, template_file, stack_name, s3_bucket, force_upload, kms_key_id, parameter_overrides, capabilities,
+           no_execute_changeset, role_arn, notification_arns, fail_on_empty_changset, tags, wait, wait_time):
+    client = boto3.client('cloudformation')
 
     try:
-        execute_command("deploy", args, template_file=template_file)
-    except OSError as ex:
-        raise UserException(str(ex))
+        change_set_name = str(uuid.uuid4()) + str(datetime.now())
+        response = client.create_change_set(
+            StackName=stack_name,
+            TemplateBody=template_file,
+            TemplateURL=s3_bucket,  # --s3-bucket
+            Parameters=[  # parameters
+                {
+                    'ParameterKey': item.split("=")[0],
+                    'ParameterValue': item.split("=")[1]
+                }
+                for item in parameter_overrides
+            ],
+            NotificationARNs=[
+                notification_arns,  # notification arns
+            ],
+            Capabilities=capabilities,
+            RoleARN=role_arn,  # role-arn
+            Tags=[  # tags
+                {
+                    'Key': item.split("=")[0],
+                    'Value': item.split("=")[1]
+                }
+                for item in tags
+            ],
+            ChangeSetName=change_set_name,
+        )
+        change_set_id = response.get("Id")
+        changes = generate_stack_changes(client, change_set_id, stack_name)
+
+        if len(changes.get("Add") + changes.get("Modify") + changes.get("Remove")) == 0 and fail_on_empty_changset:
+            click.secho("The stack {} has an empty Change Set".format(stack_name), fg="red")
+            sys.exit(1)
+
+        if changes.get("Add"):
+            secho("Resources Added")
+        for change in changes.get("Add"):
+            secho("+ " + change.get("LogicalResourceId") + " " + change.get("ResourceType"))
+
+        if changes.get("Modify"):
+            secho("Resources Modified")
+        for change in changes.get("Modify"):
+            secho("~ " + change.get("LogicalResourceId") + " " + change.get("ResourceType"))
+
+        if changes.get("Remove"):
+            secho("Resources Removed")
+        for change in changes.get("Remove"):
+            secho("- " + change.get("LogicalResourceId") + " " + change.get("ResourceType"))
+
+        if no_execute_changeset:
+            sys.exit(0)
+
+        response = client.execute_change_set(
+            ChangeSetName=change_set_name,
+            StackName=stack_name,
+        )
+
+    except ClientError as e:
+        if "AccessDeniedException" in e.response["Error"]["Message"]:
```

Comment:
Is there no specific exception that we can catch? why are we looking at a specific class of Exception within a string message?